backend/app/api/v1/endpoints/cpt/cptJobs.py

```diff
@@ -66,23 +69,35 @@ async def jobs(
         )
 
     results_df = pd.DataFrame()
+    total_dict = {}
+    total = 0
     with ProcessPoolExecutor(max_workers=cpu_count()) as executor:
         futures = {
-            executor.submit(fetch_product, product, start_date, end_date): product
+            executor.submit(
+                fetch_product, product, start_date, end_date, size, offset
+            ): product
             for product in products
         }
         for future in as_completed(futures):
             product = futures[future]
             try:
                 result = future.result()
-                results_df = pd.concat([results_df, result])
+                total_dict[product] = result["total"]
+                results_df = pd.concat([results_df, result["data"]])
             except Exception as e:
                 print(f"Error fetching data for product {product}: {e}")
 
+    # on first hit, totalJobs is 0
+    if totalJobs == 0:
+        for product in total_dict:
+            total += int(total_dict[product])
+        totalJobs = total
```

Comment:
We decided to flip this by overriding `total` with the incoming `totalJobs` rather than modifying the query parameter, and adding a comment to explain the logic.